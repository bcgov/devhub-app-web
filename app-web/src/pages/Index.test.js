import React from 'react';
import { shallow } from 'enzyme';
import Index from './index';

describe('Index Container', () => {
  test('it matches snapshot', () => {
    const data = {
      pathfinder: {
        data: {
          organization: {
            repository: {
              resources: {
                // eslint-disable-next-line
                yaml:
                  "entries:\n- {details: \"test1\" , originalSource: ../index.html, category: 'What''s Going On Now', link: 'https://github.com/BCDevOps/BCDevOps-Guide', description: 'DevOps PaaS'}\n- {details: \"test2\", originalSource: ../index.html, category: Community, link: 'https://www.yammer.com/gov.bc.ca/#/threads/inGroup?type=in_group&feedId=11241672&view=all', description: 'DevOps Commons'}\n- {details: \"test3\", originalSource: ../index.html, category: Community, link: 'https://www.meetup.com/DevOps-Commons', description: 'DevOps Commons Meetups'}\n",
              },
            },
          },
        },
      },
      allSourceDevhubGithub: {
        edges: [
          {
            node: {
              id: '1bc91db3-c484-58b5-9700-55c22406950a',
              title: 'test',
              sourceName: 'Design System',
              pagePath: '/design-system/test',
              childMarkdownRemark: {
                frontmatter: {
                  title: '',
                  description:
                    'This is a test description that will be used as the intro blurb to a card',
                },
              },
            },
          },
        ],
      },
    };
    const wrapper = shallow(<Index data={data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
