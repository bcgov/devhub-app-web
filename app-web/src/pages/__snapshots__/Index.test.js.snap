// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Container it matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome
        </h1>,
        <h2>
          <em>
            We are here to help
          </em>
        </h2>,
        <p
          className="para"
        >
          This is the front door to the developer community of the BC Government. We’re embarking on a journey to help developers learn new skills, discover resources and create amazing applications for government.
        </p>,
        <p
          className="para"
        >
          This is our first release and we’re planning to have new material added to this site every sprint (2 weeks), so please visit often to check our progress.
        </p>,
        <p
          className="para"
        >
           
          If you’d like to comment, offer a suggestion or ask a question you can find us by opening an issue in our
           
          <a
            href="https://github.com/bcgov/devhub-web/issues"
          >
            github.com
          </a>
           repository.
        </p>,
        <NavigationalItems
          navItems={
            Array [
              Object {
                "description": "Learn more about the Pathfinder initiative",
                "icon": "book",
                "link": "/learn",
                "title": "Learn",
              },
              Object {
                "description": "Register your mobile application with our in-house Signing Tool",
                "icon": "user-astronaut",
                "link": "/do",
                "title": "Do",
              },
              Object {
                "description": "This section is currently in development",
                "icon": "shipping-fast",
                "link": "/deliver",
                "title": "Deliver",
              },
            ]
          }
        />,
      ],
      "className": "main",
      "role": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome",
        },
        "ref": null,
        "rendered": "Welcome",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <em>
            We are here to help
          </em>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "We are here to help",
          },
          "ref": null,
          "rendered": "We are here to help",
          "type": "em",
        },
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is the front door to the developer community of the BC Government. We’re embarking on a journey to help developers learn new skills, discover resources and create amazing applications for government.",
          "className": "para",
        },
        "ref": null,
        "rendered": "This is the front door to the developer community of the BC Government. We’re embarking on a journey to help developers learn new skills, discover resources and create amazing applications for government.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is our first release and we’re planning to have new material added to this site every sprint (2 weeks), so please visit often to check our progress.",
          "className": "para",
        },
        "ref": null,
        "rendered": "This is our first release and we’re planning to have new material added to this site every sprint (2 weeks), so please visit often to check our progress.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "If you’d like to comment, offer a suggestion or ask a question you can find us by opening an issue in our",
            " ",
            <a
              href="https://github.com/bcgov/devhub-web/issues"
            >
              github.com
            </a>,
            " repository.",
          ],
          "className": "para",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "If you’d like to comment, offer a suggestion or ask a question you can find us by opening an issue in our",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "github.com",
              "href": "https://github.com/bcgov/devhub-web/issues",
            },
            "ref": null,
            "rendered": "github.com",
            "type": "a",
          },
          " repository.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navItems": Array [
            Object {
              "description": "Learn more about the Pathfinder initiative",
              "icon": "book",
              "link": "/learn",
              "title": "Learn",
            },
            Object {
              "description": "Register your mobile application with our in-house Signing Tool",
              "icon": "user-astronaut",
              "link": "/do",
              "title": "Do",
            },
            Object {
              "description": "This section is currently in development",
              "icon": "shipping-fast",
              "link": "/deliver",
              "title": "Deliver",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome
          </h1>,
          <h2>
            <em>
              We are here to help
            </em>
          </h2>,
          <p
            className="para"
          >
            This is the front door to the developer community of the BC Government. We’re embarking on a journey to help developers learn new skills, discover resources and create amazing applications for government.
          </p>,
          <p
            className="para"
          >
            This is our first release and we’re planning to have new material added to this site every sprint (2 weeks), so please visit often to check our progress.
          </p>,
          <p
            className="para"
          >
             
            If you’d like to comment, offer a suggestion or ask a question you can find us by opening an issue in our
             
            <a
              href="https://github.com/bcgov/devhub-web/issues"
            >
              github.com
            </a>
             repository.
          </p>,
          <NavigationalItems
            navItems={
              Array [
                Object {
                  "description": "Learn more about the Pathfinder initiative",
                  "icon": "book",
                  "link": "/learn",
                  "title": "Learn",
                },
                Object {
                  "description": "Register your mobile application with our in-house Signing Tool",
                  "icon": "user-astronaut",
                  "link": "/do",
                  "title": "Do",
                },
                Object {
                  "description": "This section is currently in development",
                  "icon": "shipping-fast",
                  "link": "/deliver",
                  "title": "Deliver",
                },
              ]
            }
          />,
        ],
        "className": "main",
        "role": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome",
          },
          "ref": null,
          "rendered": "Welcome",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <em>
              We are here to help
            </em>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "We are here to help",
            },
            "ref": null,
            "rendered": "We are here to help",
            "type": "em",
          },
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is the front door to the developer community of the BC Government. We’re embarking on a journey to help developers learn new skills, discover resources and create amazing applications for government.",
            "className": "para",
          },
          "ref": null,
          "rendered": "This is the front door to the developer community of the BC Government. We’re embarking on a journey to help developers learn new skills, discover resources and create amazing applications for government.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is our first release and we’re planning to have new material added to this site every sprint (2 weeks), so please visit often to check our progress.",
            "className": "para",
          },
          "ref": null,
          "rendered": "This is our first release and we’re planning to have new material added to this site every sprint (2 weeks), so please visit often to check our progress.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "If you’d like to comment, offer a suggestion or ask a question you can find us by opening an issue in our",
              " ",
              <a
                href="https://github.com/bcgov/devhub-web/issues"
              >
                github.com
              </a>,
              " repository.",
            ],
            "className": "para",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "If you’d like to comment, offer a suggestion or ask a question you can find us by opening an issue in our",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "github.com",
                "href": "https://github.com/bcgov/devhub-web/issues",
              },
              "ref": null,
              "rendered": "github.com",
              "type": "a",
            },
            " repository.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navItems": Array [
              Object {
                "description": "Learn more about the Pathfinder initiative",
                "icon": "book",
                "link": "/learn",
                "title": "Learn",
              },
              Object {
                "description": "Register your mobile application with our in-house Signing Tool",
                "icon": "user-astronaut",
                "link": "/do",
                "title": "Do",
              },
              Object {
                "description": "This section is currently in development",
                "icon": "shipping-fast",
                "link": "/deliver",
                "title": "Deliver",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
